ansible/roles/signal/tasks/main.yml

```diff
@@ -1,10 +1,28 @@
 ---
-- pause:
+- name: Check if signal_server_phone_number is registered
+  uri:
+    url: "{{ signal_cli_rest_api_endpoint }}/v1/receive/{{ signal_server_phone_number }}"
+  ignore_errors: true
+  register: response
+
+- name: Set error message
+  set_fact:
+    error_message: "User {{ signal_server_phone_number }} is not registered.\n"
+
+- name: Set signal_phone_number_not_registered variable
+  set_fact:
+    signal_phone_number_not_registered: "{{ response['status'] != 200 and response['json']['error'] == error_message }}"
+
+- name: Ask for captcha result
+  pause:
     prompt: "{{ lookup('template', 'enter_captcha.j2') | trim }}"
   register: captcha_prompt_result
+  when: signal_phone_number_not_registered
 
-- set_fact:
+- name: Set captcha result
+  set_fact:
     captcha: "{{ captcha_prompt_result.user_input }}"
+  when: signal_phone_number_not_registered
```

Comment:
Something I noticed while having a quick look at this: You can group multiple tasks in [blocks](https://docs.ansible.com/ansible/latest/user_guide/playbooks_blocks.html) if the tasks share certain properties. For example, in this case, you could group all tasks related to registering/verifying and use a single `when` condition on that block (instead of setting the `when` condition on every step individually.)

I’ll take a closer look in the following days.

Reply:
That's cool! I've tested it out in e7676114 and it seems to work.

Reply:
@tillprochaska Have you had the chance to have a look?